odml/dtypes.py

```diff
@@ -112,8 +113,12 @@ def set(value, dtype=None, encoding=None):
         return tuple_set(value)
     if dtype == "binary":
         return binary_set(value, encoding)
-    if type(value) in (str, unicode):
-        return str_set(value)
+    if sys.version_info > (3, 0):
```

Comment:
In such cases, I like to build a tuple of types immediately after import, for example:
```python
stringtypes = (str, bytes)
try:
    stringtypes += (basestring,)  # fails in py3
except NameError:
    pass
```
and then just test `isinstance` against `stringtypes`.
Unless you don't like using exceptions for flow control.